### ISSUE TITLE
Completion snippet to expand lonely `def` using the preceding sig

### ISSUE THREAD

**Issue by jez** [sorbet/sorbet#5444]
**NOTE** This ticket is blocked by #5443 

#### Input

[→ View on sorbet.run](https://sorbet.run/?arg=--print=parse-tree-whitequark#%23%20typed%3A%20true%0Aclass%20A%0A%20%20extend%20T%3A%3ASig%0A%20%20sig%20%7Bparams%28x%3A%20Integer%29.void%7D%0A%20%20def%0A%20%20%23%20%20%5E%20completion%0Aend)

```ruby
# typed: true
class A
  extend T::Sig
  sig {params(x: Integer).void}
  def
  #  ^ completion
end
```

#### Observed output

```
s(:begin)
editor.rb:7: unexpected token "end of file" https://srb.help/2001
     7 |end
     8 |
Errors: 1
```

#### Expected behavior

At the point where the `^ completion` comment is, it would be nice to show a completion snippet that autofills the method arity using the information from the `sig`.

It wouldn't be perfect, because Sorbet can't know whether to use positional or keyword args nor whether they should have any defaults, but it'd be really nice to have Sorbet autofill it.

Note that Sorbet already has a completion suggestion for going the other way (write the `def` first, and have Sorbet attempt to suggest the sig).

- - -
